src/platform/PICO/adc_pico.c

```diff
@@ -0,0 +1,208 @@
+}
+
+void adcInit(const adcConfig_t *config)
+{
+    adc_init();
+
+    // Set channels (ioTags) for enabled ADC inputs
+    if (config->vbat.enabled) {
+        adcOperatingConfig[ADC_BATTERY].tag = config->vbat.ioTag;
+    }
+
+    if (config->rssi.enabled) {
+        adcOperatingConfig[ADC_RSSI].tag = config->rssi.ioTag;
+    }
+
+    if (config->external1.enabled) {
+        adcOperatingConfig[ADC_EXTERNAL1].tag = config->external1.ioTag;
+    }
+
+    if (config->current.enabled) {
+        adcOperatingConfig[ADC_CURRENT].tag = config->current.ioTag;
+    }
+
+    uint mask = 0;
+
+    // loop over all possible channels and build the adcOperatingConfig to represent
+    // the set of enabled channels
+    for (int i = 0; i < ADC_SOURCE_COUNT; i++) {
+        adcOperatingConfig[i].enabled = false;
+        do {
+#ifdef USE_ADC_INTERNAL
+            if (i >= ADC_SOURCE_INTERNAL_FIRST_ID) {
+                if (i == ADC_TEMPSENSOR) {
+                    adcOperatingConfig[i].channel = PICO_ADC_INTERNAL_TEMP_CHANNEL;
+                    adcOperatingConfig[i].enabled = true;
+                }
+                break;
+            }
+#endif
+            if (!adcOperatingConfig[i].tag) {
+                break;
+            }
+
+            const int pin = DEFIO_TAG_PIN(adcOperatingConfig[i].tag);
+            const int channel = adcChannelByPin(pin);
+            if (channel >= 0) {
+                adcOperatingConfig[i].channel = channel;
+                adc_gpio_init(pin);
+                adcOperatingConfig[i].enabled = true;
+            }
+        } while (false);
+
+        if (!adcOperatingConfig[i].enabled) {
+            continue;
+        }
+
+        if (mask == 0) {
+            // select the first channel for input
+            adc_select_input(adcOperatingConfig[i].channel);
+        }
+        mask |= (1 << adcOperatingConfig[i].channel);
+    }
+
+    const uint8_t sources = BITCOUNT(mask);
+    if (sources == 0) {
+        /* don't enable the interrupt */
+        return;
+    }
+
+    adc_set_round_robin(mask);
+    adc_fifo_setup(
+        true,               // Write each completed conversion to the sample FIFO
+        false,              // Disable DMA data request (DREQ)
+        MAX(1, sources),    // IRQ asserted when at least 1 sample is present
+        false,              // We won't see the ERR bit because of 12-bit reads; disable
+        false               // Don't shift each sample to 8 bits when pushing to FIFO
+    );
+    /*
+        Sampling requires 96 cycles per sample
+        Sample as slow as possible
+    */
+    adc_set_clkdiv((65535.f + 255.f/256.f));
+
+    // --- Interrupt Setup ---
+    irq_set_exclusive_handler(ADC_IRQ_FIFO, adc_irq_handler);
+    adc_irq_set_enabled(true);
+    irq_set_enabled(ADC_IRQ_FIFO, true);
+
+    /* start the ADC in free running mode */
+    adc_run(true);
+}
+
+uint16_t adcGetValue(adcSource_e source)
+{
+    if (!adcOperatingConfig[source].enabled) {
```

Comment:
```suggestion
    if (source >= ARRAYLEN(adcOperatingConfig) || !adcOperatingConfig[source].enabled) {
```

Reply:
Although source is an enum, and the arraylength is set by that enum count. Need to cast the enum as unsigned.